quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java

```diff
@@ -174,6 +165,12 @@ private static long nextFiberId() {
      */
     @SuppressWarnings("LeakingThisInConstructor")
     public Fiber(String name, FiberScheduler scheduler, int stackSize, SuspendableCallable<V> target) {
+
+        this.traceInterrupt = SystemProperties.isEmptyOrTrue("co.paralleluniverse.fibers.traceInterrupt");
```

Comment:
Some thoughts:

1. I wonder if we're doing a bunch on unnecessary processing every time we launch a Fiber here?  It might seem like a little but it all adds up.  I think I would be in favour of some logic that determines whether it needs to re-read the system properties or not.
2. It might be clearer (and probably facilitates 1) if we encapsulated these inside a `FiberSystemProperties` class that we hold an instance of in a static volatile field.  The constructor then just grabs a reference to the volatile static field in the constructor and stores it or the values it holds in final fields of the Fiber.  You have a static method to recalculate the static volatile `FiberSystemProperties` that tests can call.
3. There might be a downside to moving away from the statics to local fields.  I think the local fields will be captured in the serialised Fiber and thus you cannot restart a process with the flags set differently and restore a Fiber to then have it verified or trace etc.  We'd need to figure out if we can reload the `FileSystemProperties` captured by the Fiber (or the values of the flags) during Fiber deserialize.

Reply:
I think the serialization issue is a very good point. I think that changing the size of the class in the same version is not a good idea at all. That's my bad. I'll put them back to statics and have an update mechanism. 
There is already a SystemProperties class for this. But there is no update mechanism when properties change.